### PR TITLE
Allow <embed> elements to be converted properly, fix blockquote format

### LINKED ISSUE #23: Blockquote
### Description of the bug

When generating the blockquote block, the output is

```
[
    [block_name] => 'quote'
    [attributes] => []
    [content] => '<blockquote><p>Lorem ipsum</p></blockquote>'
]
```

As the `blockquote` has inner html, that html should also be converted to blocks so the final result should be:

```
[
    [block_name] => 'quote'
    [attributes] => []
    [content] => '<blockquote><!-- wp:paragraph --><p>Lorem ipsum</p><!-- /wp:paragraph --></blockquote>'
]

```

### Steps To Reproduce

```
$html = '<blockquote><p>Lorem Ipsum</p></blockquote>';

$converter = new Block_Converter( $html );
$converted_blocks = $converter->convert();

var_dump($converted_blocks);
```

### Additional Information

The original output works on the Front-end, however it is producing an validation error in the editor.

### LINKED ISSUE #24: Fatal Error when encountering <embed> elements
### Description of the bug

The block converter throws an uncaught TypeError when encountering an `<embed>` element because it passes the node into `Block_Converter`'s `embed` method ([here, I believe](https://github.com/alleyinteractive/wp-block-converter/blob/2145f4bcdda261400d5c9dee7cc02aecfd1d1115/src/class-block-converter.php#L62)), which is meant to process URLs into oEmbeds and expects a string, rather than a node. Perhaps this method could simply be renamed to something that isn't a valid (or historically common but invalid) HTML element, or it could be moved to a separate class. Maybe something like `generic_embed()` would be appropriate, to contrast with `instagram_embed()` and `facbook_embed()`.

```
PHP Fatal error:  Uncaught TypeError: Alley\WP\Block_Converter\Block_Converter::embed(): Argument #1 ($url) must be of type string, DOMElement given, called in /var/www/site/wp-content/plugins/site-migration/vendor/alleyinteractive/wp-block-converter/src/class-block-converter.php on line 62 and defined in /var/www/site/wp-content/plugins/site-migration/vendor/alleyinteractive/wp-block-converter/src/class-block-converter.php:273
Stack trace:
#0 /var/www/site/wp-content/plugins/site-migration/vendor/alleyinteractive/wp-block-converter/src/class-block-converter.php(62): Alley\WP\Block_Converter\Block_Converter->embed()
```

### Steps To Reproduce

Pass content which contains an `embed` element into the block converter. example:
```
<p>Hold on to your butts</p>
<embed type="video/webm" src="/media/mr-arnold.mp4" width="250" height="200" />
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by srtfisher**
Fixes #24. Fixes #23

Will convert a string such as:

```html
<embed type="video/webm" src="/media/mr-arnold.mp4" width="250" height="200" />
```

to a block:

```html
<!-- wp:html --><embed type="video/webm" src="/media/mr-arnold.mp4" width="250" height="200"></embed><!-- /wp:html -->
```

Relates to #14. Also using this as a PR to upgrade the package's dependencies and align the actions to the organization's standards. The tests are now powered by a data provider.

### CODE REVIEW

File: tests/feature/BlockConverterTest.php [hunk 1]
```diff
@@ -9,15 +9,29 @@
...
  */
-class Test_Block_Block_Converter extends Test_Case {
+#[Group( 'block' )]
+class BlockConverterTest extends Test_Case {
+	use Prevent_Remote_Requests;
+
+	protected function setUp(): void {
+		parent::setUp();
+
+		$this->fake_request( [
```
Comment (srtfisher) [inline tests/feature/BlockConverterTest.php:29]: These were previously requests made during the test than are now being mocked.

**Comment by srtfisher** [comment]
Need to investigate why I had to modify the `fake_request()` call (upstream bug in the testing framework).

File: src/class-block-converter.php [hunk 2]
```diff
@@ -112,6 +112,34 @@ public function __call( $name, $arguments ): ?Block {
...
+	/**
+	 * Convert the children of a node to blocks.
+	 *
+	 * @param DOMNode $node The node.
+	 * @return string The children as blocks.
+	 */
+	public function convert_with_children( DOMNode $node ): string {
+		$children = '';
+
+		// Recursively convert the children of the blockquote to blocks.
```
Comment (scottnelle) [inline src/class-block-converter.php]: Blockquotes only, or might this be used for other blocks as well?

**Review by scottnelle — APPROVED** [review]
👻

**Pull request merged**